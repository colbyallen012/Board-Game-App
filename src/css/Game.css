article {
    margin: 0px;
    width: 25%;
}
img.gameCards {
    height: 150px;
}
