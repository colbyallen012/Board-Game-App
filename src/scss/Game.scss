.gameCard {
  height: 200px;
  margin: 0px;
  width: 25%;
}

img.gameCardImage {
  opacity: 1;
  transition: .5s ease;
  height: 150px;
}

.additionalGameInfo {
  transition: .5s ease;
  width: 150px;
  opacity: 0;
  transform: translate(-0%, -125%);
}

.gameCard:hover .gameCardImage {
  opacity: 0.3;
}

.gameCard:hover .additionalGameInfo {
  opacity: 1;
}

.text {
  background-color: #FFFFFF;
  color: #000000;
  font-size: 16px;
}
