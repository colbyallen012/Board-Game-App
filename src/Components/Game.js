import React, { Component } from 'react';
import '../css/Game.css'
function Game(props) {
  return (
    <div class = 'card'>
      <article>
        <img src={props.image} class ='gameCards' />
      </article>
    </div>
  )

}

export default Game;
