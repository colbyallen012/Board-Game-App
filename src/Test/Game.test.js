import React from 'react';
import ReactDOM from 'react-dom';
import Game from '../Components/Game';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const mock = [
{name: "7 Wonders",
  minPlayers: 2,
  maxPlayers: 7,
  weight: "medium",
  timeToPlay: 30,
  type: "strategy",
  features: ["card drafting","hand management"],
  image: "https://i.imgur.com/lkrz4I3.jpg",
  gameId: 1}
]

describe('Game', () => {
  let wrapper;
  
  beforeEach(() =>{
    wrapper = shallow(
      <Game
      name={mock.name}
        minPlayers={mock.minPlayers}
        maxPlayers={mock.maxPlayers}
        weight={mock.weight}
        timeToPlay={mock.timeToPlay}
        type={mock.type}
        features={mock.features}
        image={mock.image}
        id={mock.gameId}
      />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});


