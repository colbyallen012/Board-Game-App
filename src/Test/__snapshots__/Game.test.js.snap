// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should match the snapshot with all data passed in correctly 1`] = `
<div
  className="gameCard"
>
  <img
    className="gameCardImage"
  />
  <div
    className="additionalGameInfo"
  >
    <p
      className="text"
    />
    <p
      className="text"
    >
      Players: 
      -
    </p>
    <p
      className="text"
    >
      Weight: 
    </p>
    <p
      className="text"
    >
      Type: 
    </p>
  </div>
</div>
`;
